scrapers/freshmaker.py

```diff
@@ -61,34 +63,43 @@ def query_api_and_update_neo4j(self):
 
                 event.conditional_connect(event.triggered_by_advisory, advisory)
 
-                for build in fm_event['builds']:
+                for build_dict in fm_event['builds']:
                     # To handle a faulty container build in Freshmaker
-                    if not build['build_id']:
+                    if not build_dict['build_id'] or int(build_dict['build_id']) < 0:
+                        continue
+
+                    try:
+                        # The build ID obtained from Freshmaker API is actually a Koji task ID
+                        task_result = self.get_task_results(build_dict['build_id'])[0]
+                    except IndexError:
                         continue
 
-                    cb_params = dict(
-                        id_=build['id'],
-                        build_id=build['build_id'],
-                        dep_on=build['dep_on'],
-                        event_id=build['event_id'],
-                        name=build['name'],
-                        original_nvr=build['original_nvr'],
-                        rebuilt_nvr=build['rebuilt_nvr'],
-                        state=build['state'],
-                        state_name=build['state_name'],
-                        state_reason=build['state_reason'],
-                        time_submitted=timestamp_to_datetime(build['time_submitted']),
-                        type_=build['type'],
-                        type_name=build['type_name'],
-                        url=build['url']
-                    )
-                    if build['time_completed']:
-                        cb_params['time_completed'] = timestamp_to_datetime(
-                            build['time_completed'])
-                    cb = ContainerBuild.create_or_update(cb_params)[0]
-                    event.triggered_container_builds.connect(cb)
+                    # Extract the build ID from a task result
+                    xml_root = ET.fromstring(task_result['result'])
+                    build_id = xml_root.find(".//*[name='koji_builds'].//string")
```

Comment:
It worries me that `koji_builds` is plural. Have you seen one in Teiid that has more than one build? It could just be a grammar error by the developer.

Reply:
hm, that's an interesting observation. Maybe we should confirm it with someone

Reply:
I have added a TODO there and I will also create an issue, if we happen to find out that there can be multiple builds then we can change it. For the time being, I am merging this so that it doesn't block the API end-point change.

Reply:
Asked @ralphbean on IRC about this and he thinks that a task can only trigger one build. Hence, not creating an issue for this.